### DISCUSSION TITLE
Test if node will be faster

### DISCUSSION THREAD

**Discussion by jcbhmr** [Andrew-Chen-Wang/github-wiki-action#29] [General]
_Originally from Milestone https://github.com/Andrew-Chen-Wang/github-wiki-action/milestone/1_

It'd be much better if we use node since, if we add more features, then we can test them with node.js.

I also heard node is faster in general. But not too sure considering how lean this package is.

**Comment by jcbhmr** [comment]
From further research, my assumption that Node is easier to test is a bit flawed. Yes, Node.js does have more **granular** testing capabilities since you can just

```js
import { test } from "node:test"
import assert from "node:assert"
import myFunction from "../src/myFunction.js"

test("it works", () => {
  assert(myFunction(40, 2) === 42);
})
```

but with this kind of action:
1. its simple. there's no use for such a complex js package when it's just fiddling with `git` commands
2. it actually easier to test Bash since you don't need to build the JS to test it! Remember, Node.js actions **don't get a node_modules folder!**

---

as for speed, well anything is faster than Docker! Pulling and building is wayyyy slower than just `node index.js` or `bash -e script.sh`. It just comes down to which edges out the other in Bash vs. Node.js. Both are **fast enough** for almost all purposes.

---

TLDR: this github wiki stuff is too simple to need such complexity. see https://github.com/spenserblack/actions-wiki/discussions/66 for a similar discussion to this one
